mondrian/src/main/java/mondrian/olap/fun/CrossJoinFunDef.java

```diff
@@ -999,29 +999,27 @@ private static boolean checkData(
         Evaluator evaluator)
     {
         if (cnt < 0) {
-            // no measures found, use standard algorithm
-            if (measureSet.isEmpty()) {
-                Object value = evaluator.evaluateCurrent();
-                if (value != null
-                    && !(value instanceof Throwable))
-                {
-                    return true;
-                }
-            } else {
+            boolean found = false;
+            Object value = evaluator.evaluateCurrent();
+            if (value != null
+                && !(value instanceof Throwable))
+            {
+                found = true;
+            }
+            if (!measureSet.isEmpty()) {
```

Comment:
I like Matt's solution. Just make sure there is a test with a calculated member within the tuple list.